File: common/src/main/java/pers/fz/mvvm/base/BaseRecyclerViewModel.java
package pers.fz.mvvm.base;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;

import pers.fz.mvvm.bean.base.PageBean;
import pers.fz.mvvm.repository.IRepository;

/**
 * Created by fz on 2020/12/17 16:23
 * describe:
 */
public abstract class BaseRecyclerViewModel<IR extends IRepository<BV>,BV extends BaseView,T> extends BaseViewModel<IR,BV> {

    public BaseRecyclerViewModel(@NonNull Application application) {
        super(application);
    }


    public final MutableLiveData<PageBean<T>> listLiveData = new MutableLiveData<>();

    public MutableLiveData<PageBean<T>> getListLiveData() {
        return listLiveData;
    }
}

File: common/src/main/java/pers/fz/mvvm/base/kotlin/BaseRecyclerViewModel.kt
package pers.fz.mvvm.base.kotlin

import android.app.Application
import kotlinx.coroutines.flow.MutableStateFlow
import pers.fz.mvvm.base.BaseView
import pers.fz.mvvm.bean.base.PageBean

/**
 * Created by fz on 2020/12/17 16:23
 * describe:
 */
abstract class BaseRecyclerViewModel<BV : BaseView, T>(application: Application) : BaseViewModel<BV>(application) {
   open var pageMutableStateFlow = MutableStateFlow<PageBean<T>?>(null)
}